File: src/java/Gaia/model/Projeto.java
package Gaia.model;

import java.util.List;
import java.util.ArrayList;
import mpro.MproEntity.MproEntity;
import org.bson.types.ObjectId;
import org.mongodb.morphia.annotations.Reference;

/**
 *
 * @author matheus
 */
public class Projeto extends MproEntity
{

    public String Nome;
    public double AlturaPaginas;
    public double LarguraPaginas;
    @Reference
    public List<Layout> layout = new ArrayList();
    @Reference
    public List<Paginas> paginas = new ArrayList();
    @Reference
    public List<Recursos> recursos = new ArrayList();
    public String Obs;
    public String JsonEntities;

    public Projeto()
    {
    }

    public Projeto(String nome, double altura, double largura, String obs)
    {
        this.Nome = nome;
        this.AlturaPaginas = altura;
        this.LarguraPaginas = largura;
        this.Obs = obs;
    }

    public void resetAllIds () 
    {
        this.cod = new ObjectId();
        
        for (Paginas pagina : this.paginas)
        {
            pagina.resetAllIds();
        }
        
        for (Layout layout : this.layout)
        {
            layout.resetAllIds();
        }
        
        for (Recursos recurso : this.recursos)
        {
            recurso.resetAllIds();
        }
    }
    
    public void deleteDeleteds()
    {
        for (Paginas pagina : this.paginas)
        {
            pagina.deleteDeleteds();
        }
    }
}

File: web/js/gaiaModel/Projeto.js
/* 
 * Classe PROJETO
 */

function Projeto(nome, altura, largura, obs)
{
    var projEntities = null;
    this.Nome = nome;
    this.AlturaPaginas = altura;
    this.LarguraPaginas = largura;
    this.layout = new Array();
    this.paginas = new Array();
    this.recursos = new Array();
    this.Obs = obs;
    this.JsonEntities = "";
    this.cod = null;
    //this.superCod = 2147483647;

    this.ParseJsonEntities = function ()
    {
        projEntities = JSON.parse(this.JsonEntities);
    };

    this.EntitiesToJson = function ()
    {
        this.JsonEntities = JSON.stringify(projEntities);
    };

    this.getEntities = function ()
    {
        return projEntities;
    };

    this.cast = function ()
    {
        var i = 0;

        for (i = 0; i < this.layout.length; i++)
        {
            this.layout[i] = $.extend(new Layout(), this.layout[i]);
        }

        for (i = 0; i < this.paginas.length; i++)
        {
            this.paginas[i] = $.extend(new Paginas(), this.paginas[i]);
            this.paginas[i].cast();
        }

        for (i = 0; i < this.recursos.length; i++)
        {
            this.recursos[i] = $.extend(new Recursos(), this.recursos[i]);
        }
    };

}

